Traceback (most recent call last):
  File "/Users/pujatrivedi/miniconda3/envs/jupyter_book/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/pujatrivedi/miniconda3/envs/jupyter_book/lib/python3.9/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/pujatrivedi/miniconda3/envs/jupyter_book/lib/python3.9/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/pujatrivedi/miniconda3/envs/jupyter_book/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/Users/pujatrivedi/miniconda3/envs/jupyter_book/lib/python3.9/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/pujatrivedi/miniconda3/envs/jupyter_book/lib/python3.9/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/pujatrivedi/miniconda3/envs/jupyter_book/lib/python3.9/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
ad = anndata.read_h5ad(file,backed='r')
gene = ad.var
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mFileNotFoundError[0m                         Traceback (most recent call last)
Cell [0;32mIn[8], line 1[0m
[0;32m----> 1[0m ad [38;5;241m=[39m [43manndata[49m[38;5;241;43m.[39;49m[43mread_h5ad[49m[43m([49m[43mfile[49m[43m,[49m[43mbacked[49m[38;5;241;43m=[39;49m[38;5;124;43m'[39;49m[38;5;124;43mr[39;49m[38;5;124;43m'[39;49m[43m)[49m
[1;32m      2[0m gene [38;5;241m=[39m ad[38;5;241m.[39mvar

File [0;32m~/miniconda3/envs/jupyter_book/lib/python3.9/site-packages/anndata/_io/h5ad.py:197[0m, in [0;36mread_h5ad[0;34m(filename, backed, as_sparse, as_sparse_fmt, chunk_size)[0m
[1;32m    195[0m         mode [38;5;241m=[39m [38;5;124m"[39m[38;5;124mr+[39m[38;5;124m"[39m
[1;32m    196[0m     [38;5;28;01massert[39;00m mode [38;5;129;01min[39;00m {[38;5;124m"[39m[38;5;124mr[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mr+[39m[38;5;124m"[39m}
[0;32m--> 197[0m     [38;5;28;01mreturn[39;00m [43mread_h5ad_backed[49m[43m([49m[43mfilename[49m[43m,[49m[43m [49m[43mmode[49m[43m)[49m
[1;32m    199[0m [38;5;28;01mif[39;00m as_sparse_fmt [38;5;129;01mnot[39;00m [38;5;129;01min[39;00m (sparse[38;5;241m.[39mcsr_matrix, sparse[38;5;241m.[39mcsc_matrix):
[1;32m    200[0m     [38;5;28;01mraise[39;00m [38;5;167;01mNotImplementedError[39;00m(
[1;32m    201[0m         [38;5;124m"[39m[38;5;124mDense formats can only be read to CSR or CSC matrices at this time.[39m[38;5;124m"[39m
[1;32m    202[0m     )

File [0;32m~/miniconda3/envs/jupyter_book/lib/python3.9/site-packages/anndata/_io/h5ad.py:128[0m, in [0;36mread_h5ad_backed[0;34m(filename, mode)[0m
[1;32m    125[0m [38;5;28;01mdef[39;00m [38;5;21mread_h5ad_backed[39m(filename: Union[[38;5;28mstr[39m, Path], mode: Literal[[38;5;124m"[39m[38;5;124mr[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mr+[39m[38;5;124m"[39m]) [38;5;241m-[39m[38;5;241m>[39m AnnData:
[1;32m    126[0m     d [38;5;241m=[39m [38;5;28mdict[39m(filename[38;5;241m=[39mfilename, filemode[38;5;241m=[39mmode)
[0;32m--> 128[0m     f [38;5;241m=[39m [43mh5py[49m[38;5;241;43m.[39;49m[43mFile[49m[43m([49m[43mfilename[49m[43m,[49m[43m [49m[43mmode[49m[43m)[49m
[1;32m    130[0m     attributes [38;5;241m=[39m [[38;5;124m"[39m[38;5;124mobsm[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mvarm[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mobsp[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mvarp[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124muns[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mlayers[39m[38;5;124m"[39m]
[1;32m    131[0m     df_attributes [38;5;241m=[39m [[38;5;124m"[39m[38;5;124mobs[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mvar[39m[38;5;124m"[39m]

File [0;32m~/miniconda3/envs/jupyter_book/lib/python3.9/site-packages/h5py/_hl/files.py:567[0m, in [0;36mFile.__init__[0;34m(self, name, mode, driver, libver, userblock_size, swmr, rdcc_nslots, rdcc_nbytes, rdcc_w0, track_order, fs_strategy, fs_persist, fs_threshold, fs_page_size, page_buf_size, min_meta_keep, min_raw_keep, locking, alignment_threshold, alignment_interval, meta_block_size, **kwds)[0m
[1;32m    558[0m     fapl [38;5;241m=[39m make_fapl(driver, libver, rdcc_nslots, rdcc_nbytes, rdcc_w0,
[1;32m    559[0m                      locking, page_buf_size, min_meta_keep, min_raw_keep,
[1;32m    560[0m                      alignment_threshold[38;5;241m=[39malignment_threshold,
[1;32m    561[0m                      alignment_interval[38;5;241m=[39malignment_interval,
[1;32m    562[0m                      meta_block_size[38;5;241m=[39mmeta_block_size,
[1;32m    563[0m                      [38;5;241m*[39m[38;5;241m*[39mkwds)
[1;32m    564[0m     fcpl [38;5;241m=[39m make_fcpl(track_order[38;5;241m=[39mtrack_order, fs_strategy[38;5;241m=[39mfs_strategy,
[1;32m    565[0m                      fs_persist[38;5;241m=[39mfs_persist, fs_threshold[38;5;241m=[39mfs_threshold,
[1;32m    566[0m                      fs_page_size[38;5;241m=[39mfs_page_size)
[0;32m--> 567[0m     fid [38;5;241m=[39m [43mmake_fid[49m[43m([49m[43mname[49m[43m,[49m[43m [49m[43mmode[49m[43m,[49m[43m [49m[43muserblock_size[49m[43m,[49m[43m [49m[43mfapl[49m[43m,[49m[43m [49m[43mfcpl[49m[43m,[49m[43m [49m[43mswmr[49m[38;5;241;43m=[39;49m[43mswmr[49m[43m)[49m
[1;32m    569[0m [38;5;28;01mif[39;00m [38;5;28misinstance[39m(libver, [38;5;28mtuple[39m):
[1;32m    570[0m     [38;5;28mself[39m[38;5;241m.[39m_libver [38;5;241m=[39m libver

File [0;32m~/miniconda3/envs/jupyter_book/lib/python3.9/site-packages/h5py/_hl/files.py:231[0m, in [0;36mmake_fid[0;34m(name, mode, userblock_size, fapl, fcpl, swmr)[0m
[1;32m    229[0m     [38;5;28;01mif[39;00m swmr [38;5;129;01mand[39;00m swmr_support:
[1;32m    230[0m         flags [38;5;241m|[39m[38;5;241m=[39m h5f[38;5;241m.[39mACC_SWMR_READ
[0;32m--> 231[0m     fid [38;5;241m=[39m [43mh5f[49m[38;5;241;43m.[39;49m[43mopen[49m[43m([49m[43mname[49m[43m,[49m[43m [49m[43mflags[49m[43m,[49m[43m [49m[43mfapl[49m[38;5;241;43m=[39;49m[43mfapl[49m[43m)[49m
[1;32m    232[0m [38;5;28;01melif[39;00m mode [38;5;241m==[39m [38;5;124m'[39m[38;5;124mr+[39m[38;5;124m'[39m:
[1;32m    233[0m     fid [38;5;241m=[39m h5f[38;5;241m.[39mopen(name, h5f[38;5;241m.[39mACC_RDWR, fapl[38;5;241m=[39mfapl)

File [0;32mh5py/_objects.pyx:54[0m, in [0;36mh5py._objects.with_phil.wrapper[0;34m()[0m

File [0;32mh5py/_objects.pyx:55[0m, in [0;36mh5py._objects.with_phil.wrapper[0;34m()[0m

File [0;32mh5py/h5f.pyx:106[0m, in [0;36mh5py.h5f.open[0;34m()[0m

[0;31mFileNotFoundError[0m: [Errno 2] Unable to open file (unable to open file: name = 'https://allen-brain-cell-atlas.s3-us-west-2.amazonaws.com/expression_matrices/WMB-10Xv2/20230630/WMB-10Xv2-CTXsp-log2.h5ad', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)

